Versie2020 05

w-jan opened the pull request:
update code and rerun after consultation of ANB and their recommandat…

update code and rerun after consultation of ANB and their recommandations made by :

- gave lesser weight to wetter GT-groups
- drop GT-group 5
- incorporate VMM-measurepoints already used for the 'grondwaterstandsindicator'

Their recommendation to use the groundwater bodies as a stratification variable isn't applied, because this would partially counteract the spatially balanced settings.

w-jan committed:
start

w-jan committed:
resulting data for new calculation mai 2020

w-jan committed:
Changed code to
  - modify stratification scheme
  - include VMM-measurelocations used for  the 'grondwaterstandsindicator'

also adaptations to work with new released watina-package

correct an error in selection of rastercels (cat 1)

w-jan committed:
update code for new stratification scheme

w-jan committed:
update code and rerun after consultation of ANB and their recommandations made by :

- gave lesser weight to wetter GT-groups
- drop GT-group 5
- incorporate VMM-measurepoints already used for the 'grondwaterstandsindicator'

Their recommendation to use the groundwater bodies as a stratification variable isn't applied, because this would partially counteract the spatially balanced settings.

w-jan committed:
update data

w-jan committed:
update:
SCPP025 was deselected (linear effect)

florisvdh committed:
Use params for refresh_data, refresh_figures

florisvdh committed:
Add render.R with example code to render with params

florisvdh committed:
st_make_valid() moved from lwgeom to sf *

See https://github.com/r-spatial/lwgeom/blame/17523053ed3c506c2307711a042536b41baf08be/README.md#L24-L25

florisvdh commented on 00_setup.Rmd:
```diff
@@ -248,11 +277,111 @@ if (refresh_data == 2) {
                         strict =  FALSE)
   rm(output_vc)
   
+#workaround voor slecht functionerende odbc-driver (= alle meetpunten importeren)
```
can you give some feedback about `watina::get_xg3()` --> does it need changes?

w-jan replied:
Floris,
I made a report-issue here [https://github.com/inbo/watina/issues/61](https://github.com/inbo/watina/issues/61)

w-jan replied:
It's solved by installation of a version 0.3.0

florisvdh commented on 00_setup.Rmd:
```diff
@@ -248,11 +277,111 @@ if (refresh_data == 2) {
+                              lg3_lcl = rep(-1, 31))
+  
+  tubes_xg3 <- bind_rows(tubes_xg3, tubes_xg3_vmm)  
   #oplossen van UTF-probleem: strings worden in inbo-SQL-databanken opgeslagen in UTF-16, terwijl hier gewerkt wordt met UTF-8. Dit geeft een probleem bij de kable-functie
```
Current version of `watina` uses `inbodb` package which addressed some character encoding problems. See https://inbo.github.io/watina/news/index.html

w-jan replied:
Good news.
I can't currently update watina: 'package ‘watina’ is not available (for R version 3.6.1)'

florisvdh replied:
OK, will have a look at this.

florisvdh replied:
Did you use `install.packages()` or `remotes::install_github()`? See [installation instructions](https://inbo.github.io/watina/#installing-testing-and-using-the-watina-package) of watina.

It seems like a warning from `install.packages()`.

w-jan replied:
A later version is installed (not an easy operation: had to quit rstudio, new session wasn't enough).
Github (inbo/watina@42df02d). I think it's 0.3.0 (because the commit points to it, but RStudio overview  mentions still version 0.2.6.9000)

florisvdh replied:
You just need a regular install of the `master` to obtain 0.3.0; the inbo/watina@42df02d commit corresponds to 0.2.6 and does not contain the workaround of inbo/watina#56 . You can check which version you have with `packageVersion("watina")`. Or at the bottom of any rendered function documentation.

When RStudio is malbehaving, just make it shut up and/or do the maintenance steps in a pure R console :wink:

w-jan replied:
Sorry Floris,
The session info was from an old cache. I had to renew it. Now it shows: Github (inbo/watina@8f21b66)

florisvdh commented on 01_Inleiding.Rmd:
```diff
@@ -81,12 +82,16 @@ if (pdfmaken == TRUE) {
+
+    _**dat je bij het uitzetten van een nieuw punt niet op voorhand weet of het wel grondwaterafhankelijk is.**_  
+    _**dat sommige typen heel variabel zijn m.b.t. het grondwaterregime.
+_**Deze nadelen worden nu als belangrijker beoordeeld.**_  
```
```suggestion
_**Deze nadelen worden nu als belangrijker beoordeeld en de keuze is gemaakt om GT-groep 5 niet mee te nemen.**_  
```

florisvdh commented on 02_Methodiek.Rmd:
```diff
@@ -32,13 +32,34 @@ Concreet:
-Concreet: 
-    - Het aantal stratificatielagen ***(nog te beslissen)*** = `r aantal_strat `
-    - Het aantal meetpunten is gelijk per laag  **_(nog te beslissen)_** = `r minaantal_tub_group`
+- We willen daarenboven ook gestratificeerd werken volgens de 5-delige GT-groepsindeling, om zo te verzekeren dat elke GT-groep voldoende gemonitord zal worden.
```
```suggestion
- We willen daarenboven ook gestratificeerd werken volgens de GT-groepsindeling, om zo te verzekeren dat elke GT-groep voldoende gemonitord zal worden.
```

florisvdh commented on 02_Methodiek.Rmd:
```diff
@@ -32,13 +32,34 @@ Concreet:
-    - Het aantal meetpunten is gelijk per laag  **_(nog te beslissen)_** = `r minaantal_tub_group`
+- We willen daarenboven ook gestratificeerd werken volgens de 5-delige GT-groepsindeling, om zo te verzekeren dat elke GT-groep voldoende gemonitord zal worden.
+We geven niet alle GT-groepen een even groot gewicht. Vegetatietypen van GT-klasse 5 worden niet meegenomen. 
+Van deze GT-klasse is immers de grondwaterafhankelijkheid plaatsgebonden en (soms heel) variabel.
```
```suggestion
Van deze GT-klasse is immers de grondwaterafhankelijkheid plaatsgebonden en (soms heel) variabel (zie hoofdstuk \@ref(inleiding)).
```

florisvdh commented on 02_Methodiek.Rmd:
```diff
@@ -32,13 +32,34 @@ Concreet:
 
+Concreet:
+
+- Het aantal stratificatielagen  = `r aantal_strat `
```
```suggestion
- Het aantal strata  = `r aantal_strat `
```

florisvdh replied:
This turns out to be '5', is that still intended?

w-jan replied:
Theoretically it's 5, but in practice 4. I changed it to 4

florisvdh committed:
render.R: add examples with pdf_document2

florisvdh commented on 02_Methodiek.Rmd:
```diff
@@ -32,13 +32,34 @@ Concreet:
+    kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"), 
+                  full_width = TRUE, 
+                  position = "left",
+                  font_size = 9,
```
A personal note: I find the font in html tables quite small. In my experience, just using a default `kable_styling()` yields an acceptable visual result in `html_document2`.

For more narrow tables (as this one) I'd use `full_width = FALSE` or `full_width = FALSE, position = "left"`. But it's also a matter of taste (and we actually need conventions, maybe they're in `INBOmd` already)

For scroll boxes I had good results with

```r
%>% 
kable_styling(font_size = 12) %>% 
    scroll_box(height = "500px")
```

florisvdh commented on src/inlezen_tsv_files.R:
```diff
@@ -0,0 +1,19 @@
+tubes_lgl_eval
+tubes_pot_group1
+tubes_xg3
+types
```
I guess this won't work

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -185,7 +185,7 @@ if (pdfmaken == TRUE) {
 ### Berekening gemiddelde benodigde oppervlakte van een GT-groep per meetpunt
 
-We gaan hier uit van de keuze voor een gelijke verdeling van de meetpunten over de strata (= GT-groepen), namelijk we wensen `r minaantal_tub_group` meetpunten per GT-groep.
+We gaan hier uit van de ongelijke verdeling van de meetpunten over de strata (= GT-groepen), waarbij meer de drogere groepen relatief meer gewicht krijgen: voor de verdeling zi tabel \@ref(tab:GTgroepen-verdeling-over-strata).
```
```suggestion
We gaan hier uit van de ongelijke verdeling van de meetpunten over de strata (= GT-groepen), waarbij de drogere groepen relatief meer gewicht krijgen: voor de verdeling zie tabel \@ref(tab:GTgroepen-verdeling-over-strata).
```

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -351,10 +351,11 @@ Hieruit kunnen we afleiden dat de grootte van de rastercellen niet te groot geko
+<!--Enkel bij GT-groep 1 is het oppervlaktegerelateerd aantal meetpunten hoger dan het vereiste aantal. 
 Dit kan nog in een latere fase worden bijgesteld.
-Voor de overige GT-groepen stemmen de aantallen overeen met het vooropgestelde aantal.
+Voor de overige GT-groepen stemmen de aantallen overeen met het vooropgestelde aantal.-->
```
This line was commented out (could as well delete it) but actually I don't see something about the resulting number here. (Maybe it's addressed somewhere else already?)

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -406,7 +407,7 @@ if (pdfmaken == TRUE) {
 
 We kunnen nu de rastercellen weergeven die een meetpunt dienen te bevatten.
-Figuur \@ref(fig:selectie-rastercel-met-meetpunt-fig1) geeft toont de selectie voor GT-groep 1, de groep habitattypen van zeer natte bodems. Er werden 20 cellen geselecteerd, die op één na, allemaal in het oosten gelegen zijn.
+Figuur \@ref(fig:selectie-rastercel-met-meetpunt-fig1) geeft toont de selectie voor GT-groep 1, de groep habitattypen van zeer natte bodems. Er werden 11 cellen geselecteerd die allemaal in het oosten gelegen zijn.
```
```suggestion
Figuur \@ref(fig:selectie-rastercel-met-meetpunt-fig1) toont de selectie voor GT-groep 1, de groep habitattypen van zeer natte bodems. Er werden 11 cellen geselecteerd die allemaal in het oosten gelegen zijn.
```

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -406,7 +407,7 @@ if (pdfmaken == TRUE) {
 
 We kunnen nu de rastercellen weergeven die een meetpunt dienen te bevatten.
-Figuur \@ref(fig:selectie-rastercel-met-meetpunt-fig1) geeft toont de selectie voor GT-groep 1, de groep habitattypen van zeer natte bodems. Er werden 20 cellen geselecteerd, die op één na, allemaal in het oosten gelegen zijn.
+Figuur \@ref(fig:selectie-rastercel-met-meetpunt-fig1) geeft toont de selectie voor GT-groep 1, de groep habitattypen van zeer natte bodems. Er werden 11 cellen geselecteerd die allemaal in het oosten gelegen zijn.
```
> Er werden 11 cellen geselecteerd

Maybe the number could be generalized a bit with an inline R statement. But this is not needed if there's no intention to re-run/revise the bookdown in the future.

w-jan replied:
I added a bit inline R

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -823,9 +826,9 @@ if (pdfmaken == TRUE) {
 
 Tabel \@ref(tab:cat1A-table) geeft een overzicht per GT-groep van het aantal geselecteerde rastercellen met een onvoldoende aantal peilbuizen.
-Ongeacht of er peilbuizen in deze rastercellen staan, zullen er in deze cellen nog meetpunten met een peilbuis moeten uitgerust worden _(volgens het huidige voorstel)_.
+Ongeacht of er peilbuizen in deze rastercellen staan, zullen er in deze cellen nog meetpunten met een peilbuis moeten uitgerust worden.
```
```suggestion
Ongeacht of er peilbuizen in deze rastercellen staan, zullen er in deze cellen nog meetpunten met een peilbuis uitgerust moeten worden.
```

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -1364,7 +1367,7 @@ tubes_cat1 <-
              by = c("rasterid", "groupnr")) %>% 
   mutate(lastyear = ifelse(lastyear == 0, NA, lastyear)) %>% 
-  select(-c(19:23))
+  select(-c(24:28))
```
also syntaxis like `select(variablename_a:variablename_b)` is supported, may work more stable for you

w-jan replied:
OK

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -1518,12 +1524,18 @@ Van de `r nrow(tubes_menyanthes)` onderzochte meetreeksen werden in totaal `r we
 
 Het grootste deel (`r as.integer(nrow(tubes_menyanthes %>% filter(str_detect(uitspraak, pattern = "te korte tijdreeks"))) / nrow(tubes_menyanthes %>% filter(str_detect(uitspraak, pattern = "expertoordeel")|str_detect(uitspraak, pattern = "te korte tijdreeks")))*100)`%) van meetpunten die beoordeeld werden met expertoordeel (zijnde `r as.integer(nrow(tubes_menyanthes %>% filter(str_detect(uitspraa ...
 
+Voor peilbuizen van rastercellen van categorie 2 betekent het niet weerhouden van de tijdreeks dat de tijdreeks van de peilbuis niet onmiddellijk kan gebruikt worden in het meetnet / de indicator.
```
```suggestion
Voor peilbuizen van rastercellen van categorie 2 betekent het niet weerhouden van de tijdreeks dat de tijdreeks van de peilbuis niet zonder verdere bewerking gebruikt kan worden in het meetnet / de indicator.
```

florisvdh commented on 03_Uitvoering.Rmd:
````diff
@@ -1925,14 +1757,15 @@ sel_cat2_raster_bis <- sel_raster_pb %>%
 ```
 
-```{r cat2-plot-bis, fig.cap = "rastercellen met een juist voldoende aantal peilbuizen", warning= FALSE}
+```{r cat2-plot-bis, fig.cap="rastercellen met een juist voldoende aantal peilbuizen", warning= FALSE, include=FALSE}
````
`include=FALSE` will still run the code but not show its results. As this code is to make a figure, I guess that code evaluation is not needed, so `eval=FALSE` may be more convenient.

florisvdh commented on 03_Uitvoering.Rmd:
````diff
@@ -2037,9 +1873,9 @@ Het is echter opvallend dat de verschillende GT-groepen waartoe ze behoren daare
+[^4]: Dit is normaal, omdat alle cellen van deze groep in de (zand)leemstreek liggen.-->
 
-```{r cat3-plot-bis, fig.cap = "rastercellen met een overschot aan peilbuizen"}
+```{r cat3-plot-bis, fig.cap = "rastercellen met een overschot aan peilbuizen", warning= FALSE, include=FALSE}
````
same remark here on `include=FALSE`

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -2789,9 +2657,61 @@ Elke gridcel waar het plaatsen van een pb gewenst is, wordt eerst gecategoriseer
  #wegschrijven van het resultaat, omdat de berekening hiervan toch wel enkele minuten tijd vraagt.
  output_vc <- write_vc(tubes_cat1_polyg %>% st_drop_geometry(), file.path(".","data","tubes_cat1_polyg"), sorting = c("rasterid","type", "polygon_id"), strict =  FALSE)
  rm(output_vc)
+write_sf(tubes_cat1_polyg, file.path("data", "bijkomende_hokken.shp")) 
```
2 remarks:

- it's often more convenient to use *.gpkg nowadays
- I wouldn't store binary files in the git repo. You can store it in a git-ignored folder, and upload to GDrive if you would like to distribute it.

w-jan replied:
Stored it in a git-ignored folder
Added also a .gpkg-file

florisvdh commented on 01_Inleiding.Rmd:
```diff
@@ -81,12 +82,16 @@ if (pdfmaken == TRUE) {
 
 _GT-groep 5 bundelt grondwaterafhankelijke HT die zodanig breed gedefinieerd zijn dat ze op sommige locaties grondwatergevoed zijn, maar soms tot meestal niet.
```
```suggestion
_GT-groep 5 bundelt grondwaterafhankelijke HT die zodanig breed gedefinieerd zijn dat ze op sommige locaties grondwatergevoed zijn, maar soms tot meestal niet._
```

florisvdh committed:
Fix incorrect markdown title syntaxis

florisvdh reviewed (commented):
I've read through your changes, seems OK @w-jan ! :+1: 

I did not check the details of all code, but I can confirm that compilation was successful both for html and pdf! (after a fix moving from `lwgeom::st_make_valid()` to `sf::st_make_valid()`). Some smaller changes are suggested in PR #9, among others the use of parameters for refreshing data & figures (with a bit example code on how to use that).

You find specific comments further down.

As a general note, I suggest to add in the report the actual types that are represented by your final selection (and/or the extra reserve points). If I'm right, it currently is only clear that 4 groups are being represented.

A bit more explanation on the reason _why_ locations with trends are being excluded seems in place (question of @katrijnalaerts). Also I think the applied time frame (1986-2016 ?) can be stated more explicitly (and justified if needed).

Further, I have general technical comments:

- You'll need to suppress some messages and warnings that are printed here and there, e.g.:

    ```
    ## `summarise()` regrouping output by 'rasterid', 'groupnr', 'gew_aantal_meetptn', 'rankclus', 'rankclus_lastyear' (override with `.groups` argument)
    ```

    ```
    ## Using `n` as weighting variable
    ## ℹ Quiet this message with `wt = n` or count rows with `wt = 1`
    ```

- Scroll boxes have an extremely small font in the HTML, I suggest to use font size 12 in that case.

- At the end of the document, a map appears '_Meetpunten van categorie 1 waar nog een peilbuis nodig is_'. I guess it is out of place there and/or may need to be set to `eval=FALSE`? Somehow, this part on "Selecteren van potentiële geschikte habitatvlekken voor rastercellen van categorie 1" feels a bit odd to put as the final 'concluding' paragraph. The one before, on "Samenvatting selectie peilbuizen", appears more appropriate for that.

- The PDF did compile (`pdf_document2`), but needs more work. I suggest to first use `INBOmd::inbo_rapport` as the output format and see from there what still needs to be fixed. I will try to assist where needed.

- To enhance reproducibility, I suggest to do the following:

     1. Update all your packages (see [here](https://inbo.github.io/tutorials/installation/user/user_install_r/#na-elke-upgrade) for CRAN packages). For `watina` & `n2khab`, best use the latest master.
    2. Run `renv::init()` in the RStudio console. (see [here](https://rstudio.github.io/renv/articles/renv.html) for a bit general info on this)
    3. Commit all newly created files. Then, new sessions will use a dedicated library of locally stored (fixed) package versions. Other users can get it too with `renv::restore()`.

The rest is mainly about repo & code management:

- `_03test_Uitvoering.Rmd`: if this file is not useful anymore: delete it and commit the deletion. That may apply to other files as well (?). The files will still be accessible from git history; it's just good practice to have finalized commits correspond to cleaned-up states (i.e. before merging / releasing). That also applies to obsoleted code and markdown that you commented out.

- Under the `src` folder, several R/Rmd files have been added. It would be good to give some guidance on this by documenting the repo structure in `README.md`. Are they all related to the work on 'droogtemeetnet'? If some don't fit in this repo, it could also be better to move some to a separate GitHub repo.

- After compiling the html my repo had gotten 'unclean' with many modified files:

```bash
$ git status
On branch version2020_05_floris
Your branch is up-to-date with 'origin/version2020_05_floris'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   data/bijkomende_hokken.dbf
	modified:   data/bijkomende_hokken.prj
	modified:   data/bijkomende_hokken.shp
	modified:   data/bijkomende_hokken.shx
	modified:   data/tubes_cat123_run1.tsv
	modified:   data/tubes_cat123_run1.yml
	modified:   data/tubes_cat1_polyg.tsv
	modified:   data/tubes_cat1_polyg.yml
	modified:   data/tubes_cat2_bis.tsv
	modified:   data/tubes_cat2_bis.yml
	modified:   data/tubes_cat3_bis.tsv
	modified:   data/tubes_cat3_bis.yml
	modified:   data/tubes_cat3_grts.tsv
	modified:   data/tubes_cat3_grts.yml
	modified:   data/tubes_cat3_grts_reserve.tsv
	modified:   data/tubes_cat3_grts_reserve.yml
	modified:   data/tubes_reserve.tsv
	modified:   data/tubes_reserve.yml
	modified:   data/tubes_selected.tsv
	modified:   data/tubes_selected.yml

no changes added to commit (use "git add" and/or "git commit -a")

```

Regarding the vc-files in `data/`, perhaps they were not yet committed (and should be)? Anyway, the changes that I got in the vc-data can be inspected in branch `version2020_05_floris_data`. This branch can be deleted afterwards (unless you're sure they should be merged indeed).

- After rendering the PDF, I saw more changes, see below. You should ignore `*.pdf`, `*.tex`, `*.log` and `book/` in `.gitignore`. Preferrably those files are also discarded from the repo history (they appear to have been committed already) to keep the latter small - I could help with that part once everything is finished.

```bash
$ git status
On branch version2020_05_floris
Your branch is up-to-date with 'origin/version2020_05_floris'.
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   03_Uitvoering.Rmd
	modified:   _book/droogtemeetnet_ontwerp.pdf
	modified:   _book/droogtemeetnet_ontwerp.tex
	modified:   data/bijkomende_hokken.dbf
	modified:   data/bijkomende_hokken.prj
	modified:   data/bijkomende_hokken.shp
	modified:   data/bijkomende_hokken.shx
	modified:   data/tubes_cat123_run1.tsv
	modified:   data/tubes_cat123_run1.yml
	modified:   data/tubes_cat1_polyg.tsv
	modified:   data/tubes_cat1_polyg.yml
	modified:   data/tubes_cat2_bis.tsv
	modified:   data/tubes_cat2_bis.yml
	modified:   data/tubes_cat3_bis.tsv
	modified:   data/tubes_cat3_bis.yml
	modified:   data/tubes_cat3_grts.tsv
	modified:   data/tubes_cat3_grts.yml
	modified:   data/tubes_cat3_grts_reserve.tsv
	modified:   data/tubes_cat3_grts_reserve.yml
	modified:   data/tubes_reserve.tsv
	modified:   data/tubes_reserve.yml
	modified:   data/tubes_selected.tsv
	modified:   data/tubes_selected.yml
	modified:   droogtemeetnet_ontwerp.log
	deleted:    droogtemeetnet_ontwerp.pdf
	deleted:    droogtemeetnet_ontwerp.tex

no changes added to commit (use "git add" and/or "git commit -a")
```

- In the above output, you can also notice that a shapefile is being tracked in the repo and marked as changed (probably just because it is written again). I suggest not to store binary files in order to keep the repo small (and because versioning them in a git repo has less meaning if they change each time they're written, without the actual data being changed). You could apply a solution as you already did for other binary files. It can e.g. be written in a git-ignored folder (and uploaded to GDrive for distribution if useful).

    Below are the commits where the shapefile has changed already. If desired, I could trash the shapefile at a later stage.

```bash
1f54d53 (2020-05-20 13:00:55 +0200) update data

M       data/bijkomende_hokken.shp
ea25456 (2020-05-15 15:05:01 +0200) resulting data for new calculation mai 2020

M       data/bijkomende_hokken.shp
d25b512 (2020-05-11 16:26:32 +0200) start

A       data/bijkomende_hokken.shp
```

These are the places where the shapefile occurs in code:

```bash
./03_Uitvoering.Rmd:2660:write_sf(tubes_cat1_polyg, file.path("data", "bijkomende_hokken.shp")) 
./droogtemeetnet_script.R:2268:write_sf(tubes_cat1_polyg, file.path("data", "bijkomende_hokken.shp"))
```

florisvdh commented:
After talking with @katrijnalaerts it appears that the following paragraphs cause some confusion (under 'Tijdsreeksanalyse') and would benefit from clearer wording / explanation:

> Het tweede doel is in feite een bijkomend~e~ selectiecriterium. Indien uit een tijdreeksanalyse blijkt dat de waterstanden van een meetpunt systematisch dalen of stijgen, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. Met een systematische daling of stijging bedoelen we een lineaire trend die niet door externe factoren (opstuwen of graven greppels, waterwinning) kan verklaard worden. Indien een tijdreeks een lineaire trend vertoont, wordt het meetpunt als momenteel ongeschikt beschouwd.

> Een meetreeks werd weerhouden als er een tijdsmodel kon voor samengesteld worden dat minstens 66% van de variatie in de waargenomen peilen (cfr. EVP, explained variance percentage, in de Menyanthes-software) kon verklaren en ook geen significante lineaire trend vertoont. Bij het tijdsmodel werden uitsluitend neerslag- en evapotranspiratiedata als verklarende reeksen gebruikt.
>
> Minstens twee derden van het peilverloop dient dus verklaard te kunnen worden door het weer. Eén derde kan/mag door andere factoren beïnvloed worden (bijv. het naburige peil van een waterloop).

Actually it is still unclear what kind of 'trend' is meant in:

> en ook geen significante lineaire trend vertoont

- is it the _remaining_ trend after taking into account precipitation & evapotranspiration? Hence a trend, caused by other factors such as ditches etc. But this seems to contradict with:
> een lineaire trend die _**niet**_ door externe factoren (opstuwen of graven greppels, waterwinning) kan verklaard worden.
- so is it the overall trend? But then you run the risk of discarding systematically ongoing drying or wetting processes from the selected locations.

If I remember correctly, we wanted to avoid cases where _other factors than climate_ caused systematic changes, right? That would make that detected changes _during_ the considered timeframe can be attributed to climate, and that this can be expected for _future_ ones too.

Or do you mean that you checked specific elements in the neighbourhood of locations, to see whether you can explain certain non-climatic changes or trends?

florisvdh commented:
Regarding my earlier comment,

>  add in the report the actual types that are represented by your final selection (and/or the extra reserve points)

below is some draft code that I used to make a few csv files - you may want to recycle parts of it. Note that it is only about the currently existing locations. Hopefully I used the right objects - otherwise it needs more modification.

```r
types <- 
  read_types(lang = "nl") %>% 
  select(type, type_name, type_shortname, typeclass_name)

tubes_selected_types <- 
  tubes_selected %>% 
  as_tibble %>% 
  filter(selectie == 1) %>% 
  select(loc_code, x, y, 
         target_groupnr = groupnr) %>% 
  left_join(tubes_hab %>% 
              select(loc_code, 
                     description,
                     type, 
                     certain, 
                     phab, 
                     groupnr,
                     group_name = typegroup_name), 
            by = "loc_code") %>% 
  left_join(types, by = "type")

tubes_selected_types %>%
  write_csv("tubes_selected_types.csv")

tubes_selected_types %>% 
  count(target_groupnr, groupnr, type, type_shortname) %>% 
  write_csv("tubes_selected_types_stats.csv")

tubes_selected_types %>% 
  count(target_groupnr, groupnr, typeclass_name) %>% 
  write_csv("tubes_selected_typeclasses_stats.csv")
```

florisvdh committed:
Table selected tubes: drop column 'selectie'

florisvdh committed:
Language fix

w-jan committed:
export some results (tubes_selected and tubes_reserve) as shape-files

store all GIS-data in a git-ignored map

w-jan committed:
Merge branch 'Versie2020_05' into version2020_05_floris

w-jan committed:
Merge pull request #9 from inbo/version2020_05_floris

A few fixes and using parameters

w-jan committed:
update and complete rerun after review Floris

w-jan committed:
update run
@floris: good news, now i get also 58 tubes that directly can be applied

w-jan committed:
adding more type of files to ignore (_book, *.pdf and *.tex)

w-jan committed:
Update 02_Methodiek.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
Update 02_Methodiek.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
data clean up

w-jan committed:
Update 03_Uitvoering.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
Update 03_Uitvoering.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
small text editing

w-jan committed:
Merge branch 'Versie2020_05' of https://github.com/inbo/n2khab-droogtemeetnet into Versie2020_05

w-jan committed:
Some text editing

w-jan committed:
Update 03_Uitvoering.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
Update 03_Uitvoering.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
updates suggested by Floris VDH (many thanks)

w-jan committed:
Merge branch 'Versie2020_05' of https://github.com/inbo/n2khab-droogtemeetnet into Versie2020_05

w-jan commented:
> Regarding my earlier comment,
> 
> > add in the report the actual types that are represented by your final selection (and/or the extra reserve points)
> 
> below is some draft code that I used to make a few csv files - you may want to recycle parts of it. Note that it is only about the currently existing locations. Hopefully I used the right objects - otherwise it needs more modification.
> 
> ```r
> types <- 
>   read_types(lang = "nl") %>% 
>   select(type, type_name, type_shortname, typeclass_name)
> 
> tubes_selected_types <- 
>   tubes_selected %>% 
>   as_tibble %>% 
>   filter(selectie == 1) %>% 
>   select(loc_code, x, y, 
>          target_groupnr = groupnr) %>% 
>   left_join(tubes_hab %>% 
>               select(loc_code, 
>                      description,
>                      type, 
>                      certain, 
>                      phab, 
>                      groupnr,
>                      group_name = typegroup_name), 
>             by = "loc_code") %>% 
>   left_join(types, by = "type")
> 
> tubes_selected_types %>%
>   write_csv("tubes_selected_types.csv")
> 
> tubes_selected_types %>% 
>   count(target_groupnr, groupnr, type, type_shortname) %>% 
>   write_csv("tubes_selected_types_stats.csv")
> 
> tubes_selected_types %>% 
>   count(target_groupnr, groupnr, typeclass_name) %>% 
>   write_csv("tubes_selected_typeclasses_stats.csv")
> ```

Integrated this code in the Rmd-document

w-jan commented:
> After talking with @katrijnalaerts it appears that the following paragraphs cause some confusion (under 'Tijdsreeksanalyse') and would benefit from clearer wording / explanation:
> 
> > Het tweede doel is in feite een bijkomend~e~ selectiecriterium. Indien uit een tijdreeksanalyse blijkt dat de waterstanden van een meetpunt systematisch dalen of stijgen, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. Met een systematische daling of stijging bedoelen we een lineaire trend die niet door externe factoren (opstuwen of graven greppels, waterwinning) kan verklaard worden. Indien een tijdreeks een lineaire trend vertoont, wordt het meetpunt als momenteel ongeschikt beschouwd.
> 
> > Een meetreeks werd weerhouden als er een tijdsmodel kon voor samengesteld worden dat minstens 66% van de variatie in de waargenomen peilen (cfr. EVP, explained variance percentage, in de Menyanthes-software) kon verklaren en ook geen significante lineaire trend vertoont. Bij het tijdsmodel werden uitsluitend neerslag- en evapotranspiratiedata als verklarende reeksen gebruikt.
> > Minstens twee derden van het peilverloop dient dus verklaard te kunnen worden door het weer. Eén derde kan/mag door andere factoren beïnvloed worden (bijv. het naburige peil van een waterloop).
> 
> Actually it is still unclear what kind of 'trend' is meant in:
> 
> > en ook geen significante lineaire trend vertoont
> 
> * is it the _remaining_ trend after taking into account precipitation & evapotranspiration? Hence a trend, caused by other factors such as ditches etc. But this seems to contradict with:
> 
> > een lineaire trend die _**niet**_ door externe factoren (opstuwen of graven greppels, waterwinning) kan verklaard worden.
> 
> * so is it the overall trend? But then you run the risk of discarding systematically ongoing drying or wetting processes from the selected locations.
> 
> If I remember correctly, we wanted to avoid cases where _other factors than climate_ caused systematic changes, right? That would make that detected changes _during_ the considered timeframe can be attributed to climate, and that this can be expected for _future_ ones too.
> 
> Or do you mean that you checked specific elements in the neighbourhood of locations, to see whether you can explain certain non-climatic changes or trends?

Thanks for pointing this.
I made some changes that hopefully clarify this part of the selection procedure

florisvdh commented on .gitignore:
```diff
@@ -51,7 +51,10 @@ rsconnect/
 _bookdown_files/
+_book/
 *.html
+*.pdf
+*.tex
```
also to be added: `*.log`

w-jan replied:
done

florisvdh commented on 00_setup.Rmd:
```diff
@@ -117,8 +115,8 @@ if (refresh_data == 2) {
-if (refresh_data == 2) {
-  habmap_terr <- read_habitatmap_terr()
+if (params$refresh_data == 2) {
+  habmap_terr <- read_habitatmap_terr(file = "20_processed/habitatmap_terr/habitatmap_terr_inclbos.gpkg")
```
It appears that you've made a variant of `habitatmap_terr` for some (undoubtedly good) reason - but is it explained in the text?

w-jan replied:
good point ! I was forgotten to documentate. Is now done

florisvdh commented on 00_setup.Rmd:
```diff
@@ -277,107 +277,16 @@ if (refresh_data == 2) {
-      tbl(con, "vwDimMeetpunt") %>%
-                   select(loc_wid = .data$MeetpuntWID,
-                          loc_code = .data$MeetpuntCode)
+  watina2 <- inbodb::connect_inbo_dbase("W0002_00_Watina")  
```
You can safely use `connect_watina()` from the `watina` package if it is at version 0.3.0.

w-jan replied:
OK. I trust it ;-)

florisvdh commented on 00_setup.Rmd:
```diff
@@ -277,107 +277,16 @@ if (refresh_data == 2) {
-                          loc_code = .data$MeetpuntCode)
+  watina2 <- inbodb::connect_inbo_dbase("W0002_00_Watina")  
+  tubes_xg3 <- tubes_hab %>% 
+    get_xg3(watina2, startyear = year(now()) - 18, endyear = 2016, vert_crs = "local",
```
> `startyear = year(now()) - 18, endyear = 2016`

Note that this timespan will shorten in the future. Should `startyear` be hardcoded if the timespan for selecting locations is intended to remain fixed?

w-jan replied:
Again good point. 
This frame is used in the selection-procedure. Tubes with a 'recent' lg3 are more favourable than tubes with only older ones.
I made changes to make it in line with the text (from 2000 unwards).
I'll modify the text also : not '2000', but 'the past 20 years'

florisvdh commented on 00_setup.Rmd:
```diff
@@ -395,143 +304,147 @@ get_xg3_temp <- function(locs,
   
   
   DBI::dbDisconnect(watina)
-  
+  DBI::dbDisconnect(watina2)
```
to be cleaned

w-jan replied:
done

florisvdh commented on 00_setup.Rmd:
```diff
@@ -395,143 +304,147 @@ get_xg3_temp <- function(locs,
-      assertthat::assert_that(all(xg3_type %in%
-                        c("L", "H", "V")),
-                  msg = "You specified at least one unknown xg3_type.")
+  filter_xg3_lok <- function(data,
```
If this and below functions, or parts of them, are interesting to share with others (I didn't go into them), best consider making a pull request in the watina repo.

w-jan replied:
I needed the use these watina-functions locally, because otherwise i became some errors eg. 'cannot open file 'Boundary_Crossing_Time.txt': Permission denied'
The content of the functions remained the same (<Watina version 0.3.0)

florisvdh replied:
I see. It seems to have to do with the KSgeneral package. watina calls `KSgeneral::disc_ks_test()`, which calls `KSgeneral::ks_c_cdf_Rcpp()`. For the latter to work, `disc_ks_test()` [writes](https://github.com/raymondtsr/ksgeneral/blob/master/R/disc_ks_test.R#L182) a file "Boundary_Crossing_Time.txt". It  appears this doesn't work on your PC.

Now I remember we already met this - #6.

w-jan replied:
Yes indeed, that's the problem (on my laptop)

florisvdh commented on 01_Inleiding.Rmd:
```diff
@@ -80,25 +81,22 @@ if (pdfmaken == TRUE) {
-_**Deze nadelen worden nu als belangrijker beoordeeld.**_  
 De GT-groepen lenen zich heel goed om een stratificatie uit te voeren op basis van de verwachte responssnelheid van een locatie op droogte. 
 De permanent gevoede locaties (GT-groep 1) zullen naar verwachting trager reageren dan de tijdelijk of zwak gevoede locaties. 
+GT-groep 5 bundelt grondwaterafhankelijke HT die zodanig breed gedefinieerd zijn dat ze op sommige locaties grondwatergevoed zijn, maar soms tot meestal niet.
+Het voordeel van deze groep is dat ze vermoedelijk het snelst op droogte zal reageren.
+Ze houdt echter ook nadelen in, nl. dat je bij het uitzetten van een nieuw punt niet op voorhand weet of het wel grondwaterafhankelijk is en dat sommige typen heel variabel zijn m.b.t. het grondwaterregime.
+We oordelen dat deze nadelen groter zijn dan het mogelijke voordeel van een snelle signalering.
+De GT-groep 5 wordt daarom niet meegenomen in de verdere analyse.
```
:+1:

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -1524,11 +1534,13 @@ Van de `r nrow(tubes_menyanthes)` onderzochte meetreeksen werden in totaal `r we
 Aangezien er geen andere peilbuizen zijn om in vervanging te voorzien, is het uitbreiden van de tijdreeks de beste kans om voor deze rastercel ooit aan een voldoend aantal geschikte tijdreeksen te komen. 
 
-Voor peilbuizen van rastercellen van categorie 3 
+Voor peilbuizen van rastercellen van categorie 3 betekent het weerhouden van de tijdreeks nog niet automatisch tot een selectie. 
```
```suggestion
Voor peilbuizen van rastercellen van categorie 3 leidt het weerhouden van de tijdreeks nog niet automatisch tot een selectie. 
```

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -1971,16 +1983,16 @@ if (nrow(tubes_cat3_bis %>% filter(selectie == 0)) > 0 | nrow(tubes_cat2_bis %>%
+          dplyr::select(loc_code, nryears, firstyear, lastyear, selectie) , 
         tubes_cat2_bis %>% 
-          select(loc_code, nryears, firstyear, lastyear, selectie) , 
+          dplyr::select(loc_code, nryears, firstyear, lastyear, selectie) , 
```
I wonder why you added `dplyr::` everywhere. Was `select()` masked by another package? In such case, the order of loading packages in setup.Rmd is the key.

But of course it's safe to keep it this way.

w-jan replied:
I know but at a given moment the R-code was interrupted because of the wrong select-function (also pull is a delicate one)  and I didn't want to start all over again. 
It's safer, but probably redundant.

florisvdh replied:
I think the conflicts are due to `raster::select()` and `git2r::pull()`. If you load dplyr or tidyverse at the end of chunk library(...) statements in index.Rmd, you shouldn't have problems (then dplyr::xxx will mask the other ones).

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -2668,16 +2561,158 @@ Elke gridcel waar het plaatsen van een pb gewenst is, wordt eerst gecategoriseer
+    kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"),
+                full_width = T,
+                position = "left",
+                font_size = 8,
```
I guess font size will need to be larger for scroll boxes (see general comments of my previous review). I didn't recompile to check.

w-jan replied:
changed it to 12. It seems ok so.

florisvdh commented on 03_Uitvoering.Rmd:
````diff
@@ -2668,16 +2561,158 @@ Elke gridcel waar het plaatsen van een pb gewenst is, wordt eerst gecategoriseer
+
+```{r tubes-as-shape, warning=FALSE}
+
+write_sf(tubes_selected_sf %>%  
+           select(-obswell_statecode, -loc_validitycode) %>% 
+           rename(watinac = "loc_code"), 
+         driver = "ESRI Shapefile", file.path("data","GIS", "tubes_selected_2020_05.shp"))
+
+write_sf(tubes_reserve %>% 
+           as_points() %>%  
+           select(-obswell_statecode, -loc_validitycode) %>% 
+           rename(watinac = "loc_code"), 
+         file.path("data","GIS", "tubes_reserve_2020_05.shp" ))
````
It's not bad to use shapefiles, but it's better to use the geopackage format. Single-file, open, better capabilities.

w-jan replied:
I added the files in the geopackage-format.

florisvdh replied:
Superb.

florisvdh reviewed (commented):
Fine updates @w-jan :+1: ! From the [previous](https://github.com/inbo/n2khab-droogtemeetnet/pull/8#pullrequestreview-437243715) general comments, I think these (may) still need attention - the first two concern textual explanation:

- > A bit more explanation on the reason _why_ locations with trends are being excluded seems in place (question of @katrijnalaerts).
    - see also the extra [comment](https://github.com/inbo/n2khab-droogtemeetnet/pull/8#issuecomment-649669675) on this
- > Also I think the applied time frame (1986-2016 ?) can be stated more explicitly (and justified if needed).
- > Scroll boxes have an extremely small font in the HTML, I suggest to use font size 12 in that case.
- Make (more) reproducible with `renv` after having all packages up to date

w-jan committed:
02_Methodiek:
Explain better how is dealt with linear trends

03_Uitvoering:
Add some overview tables with results:
how GW-types are divisioned into habitattype/rbbs and habitatgroups

florisvdh commented on 02_Methodiek.Rmd:
```diff
@@ -252,11 +252,12 @@ In dit project willen we deze analysetechniek toepassen op grondwatermeetreeksen
-Het tweede doel is in feite een bijkomende selectiecriterium. Indien uit een tijdreeksanalyse blijkt dat de waterstanden van een meetpunt systematisch dalen of stijgen, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. 
-Met een systematische daling of stijging bedoelen we een lineaire trend die niet door externe factoren (opstuwen of graven greppels, waterwinning) kan verklaard worden. 
-Indien een tijdreeks een lineaire trend vertoont, wordt het meetpunt als momenteel ongeschikt beschouwd.
+Het tweede doel is in feite een bijkomende selectiecriterium. Indien uit een tijdreeksanalyse blijkt dat, na het in rekening brengen met zoveel mogelijk factoren (maar in de praktijk enkel met het weer), de waterstanden van een meetpunt geleidelijkaan dalen of stijgen, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. 
```
```suggestion
Het tweede doel is in feite een bijkomend selectiecriterium. Indien uit een tijdreeksanalyse blijkt dat, na rekening te houden met zoveel mogelijk factoren (maar in de praktijk enkel met het weer), de waterstanden van een meetpunt geleidelijk aan dalen of stijgen, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. 
```

florisvdh replied:
> na rekening te houden met zoveel mogelijk factoren (maar in de praktijk enkel met het weer), de waterstanden van een meetpunt geleidelijk aan dalen of stijgen

"de waterstanden van een meetpunt": in deze context beter vervangen door "de niet door het weer verklaarde component van de waterstanden van een meetpunt"?

Daarmee maak je ook duidelijker dat de _waterstanden_ an sich _wel_ een trend mogen vertonen, zolang die maar verklaard zijn door een trend in het weer. Je hebt het dus over een partiële trend (na isolatie van andere, eveneens tijdsgebonden effecten). Daarin zit volgens mij de verwarring voor de lezer, misschien kan dat nog explicieter.

w-jan replied:
Thanks for your help to clarify this.

w-jan committed:
Update 02_Methodiek.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
Update 03_Uitvoering.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
general:
use tibble as standard data form
set font size to 12 pts

00_setup:
get_xg3: screen the latest 20 years

02_Methodiek:
clarify the procedure to evaluate a time series

03_Uitvoering:
Add the reference for the used habitatmap
Save all the gis-files also as a geopackage

w-jan committed:
New run

w-jan committed:
Merge branch 'Versie2020_05' of https://github.com/inbo/n2khab-droogtemeetnet into Versie2020_05

w-jan committed:
cleaning source files

w-jan committed:
cleaning source files and gitignore

w-jan committed:
renv files

w-jan committed:
Merge branch 'Versie2020_05' of https://github.com/inbo/n2khab-droogtemeetnet into Versie2020_05

w-jan committed:
update code-file to make importfiles for Menyanthes

w-jan committed:
reorganising files

w-jan committed:
supplementary update

florisvdh commented on 02_Methodiek.Rmd:
```diff
@@ -250,11 +250,13 @@ In dit project willen we deze analysetechniek toepassen op grondwatermeetreeksen
 
-Het tweede doel is in feite een bijkomend selectiecriterium. Indien uit een tijdreeksanalyse blijkt dat, na rekening te houden met zoveel mogelijk factoren (maar in de praktijk enkel met het weer), de waterstanden van een meetpunt geleidelijk aan dalen of stijgen, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. 
+Het tweede doel is in feite een bijkomend selectiecriterium. 
+Indien uit een tijdreeksanalyse blijkt dat, na rekening te houden met zoveel mogelijk factoren (maar in de praktijk enkel met het weer), de niet door het weer verklaarde component van de waterstanden van een meetpunt geleidelijk aan dalen of stijgen, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. 
```
```suggestion
Indien uit een tijdreeksanalyse blijkt dat, na rekening te houden met zoveel mogelijk factoren (maar in de praktijk enkel met het weer), de niet door het weer verklaarde component van de waterstanden van een meetpunt geleidelijk aan daalt of stijgt, leent dat meetpunt zich niet goed voor het berekenen van een droogte-indicator. 
```

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -60,14 +59,13 @@ check <- raster_meetnet_poly %>% st_drop_geometry() %>% count(rasterid) %>% filt
 
 Op basis van deze rasterlaag maken we een overlay met de verspreiding van de vijf grondwatergroepen volgens de habitatkaart (figuur \@ref(fig:overlay-habitatkaart-GRTS-plot)). 
-
+We deden hiervoor een beroep op de gestandaardiseerde versie van de habitatkaart ([bron](https://zenodo.org/record/3355193)). 
```
Best refereren als een referentie met DOI, zoals het ook op Zenodo te zien is (en als BibTeX kan worden getoond/gedownload). Zoals het ook voor habitatmap_terr is gebeurd.

florisvdh replied:
Je URL refereert nog naar de 1e versie (2018_v1); ik weet niet welke je precies hebt gebruikt, maar n2khab verwacht default versie 2 (2018_v2) dus het zal die wel moeten zijn denk ik. Maar gebruik steeds een DOI-link: https://doi.org/....

w-jan replied:
OK, heb de link naar versie 2 gegeven

florisvdh commented on 03_Uitvoering.Rmd:
```diff
@@ -60,14 +59,13 @@ check <- raster_meetnet_poly %>% st_drop_geometry() %>% count(rasterid) %>% filt
 Op basis van deze rasterlaag maken we een overlay met de verspreiding van de vijf grondwatergroepen volgens de habitatkaart (figuur \@ref(fig:overlay-habitatkaart-GRTS-plot)). 
-
+We deden hiervoor een beroep op de gestandaardiseerde versie van de habitatkaart ([bron](https://zenodo.org/record/3355193)). 
+In deze versie werden de habitatvlekken geïnterpreteerd: uitsluitend vlekken met een terrestrisch habitattype of een rbb werden behouden. 
```
```suggestion
We maakten hiervan een afgeleide databron waarin de habitatvlekken werden geïnterpreteerd: uitsluitend vlekken met een terrestrisch habitattype of een rbb werden behouden. 
```

florisvdh reviewed (commented):
Good job @w-jan !

w-jan committed:
Update 02_Methodiek.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
Update 03_Uitvoering.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
correct some dataset references

w-jan committed:
text update

Merge branch 'Versie2020_05' of https://github.com/inbo/n2khab-droogtemeetnet into Versie2020_05

# Conflicts:
#	03_Uitvoering.Rmd

w-jan committed:
small literature update

florisvdh reviewed (commented):
:+1: 

`renv::restore()` did a good job to reproduce your environment here!

I then tried to recompile (since last time was some weeks ago). Got this error:

```
Quitting from lines 169-278 (droogtemeetnet_ontwerp.Rmd) 
Error: Cannot open "/media/floris/DATA/PROJECTS/09685_NatuurlijkMilieu/160 Bewerkingen en resultaat/Repos_en_data/n2khab-preprocessing/n2khab_data/20_processed/habitatmap_terr/habitatmap_terr_inclbos.gpkg"; The file doesn't seem to exist.
Please delete droogtemeetnet_ontwerp.Rmd after you finish debugging the error.
Execution halted

Exited with status 1.
```

Meaning: you'll need to distribute this gpkg file and set a better (data) path in `read_habitatmap_terr()`, using its `file` and `path` arguments. This occurs in two places:

```sh
./00_setup.Rmd:119:  habmap_terr <- read_habitatmap_terr(file = "20_processed/habitatmap_terr/habitatmap_terr_inclbos.gpkg")
./src/meetnet/droogtemeetnet_script.R:98:  habmap_terr <- read_habitatmap_terr(file = "20_processed/habitatmap_terr/habitatmap_terr_inclbos.gpkg")
```

w-jan committed:
add missing download link for
habitatmap_terr_inclbos.gpkg

w-jan commented:
> `renv::restore()` did a good job to reproduce your environment here!
> 
> I then tried to recompile (since last time was some weeks ago). Got this error:
> 
> ```
> Quitting from lines 169-278 (droogtemeetnet_ontwerp.Rmd) 
> Error: Cannot open "/media/floris/DATA/PROJECTS/09685_NatuurlijkMilieu/160 Bewerkingen en resultaat/Repos_en_data/n2khab-preprocessing/n2khab_data/20_processed/habitatmap_terr/habitatmap_terr_inclbos.gpkg"; The file doesn't seem to exist.
> Please delete droogtemeetnet_ontwerp.Rmd after you finish debugging the error.
> Execution halted
> 
> Exited with status 1.
> ```
> 
> Meaning: you'll need to distribute this gpkg file and set a better (data) path in `read_habitatmap_terr()`, using its `file` and `path` arguments. This occurs in two places:
> 
> ```shell
> ./00_setup.Rmd:119:  habmap_terr <- read_habitatmap_terr(file = "20_processed/habitatmap_terr/habitatmap_terr_inclbos.gpkg")
> ./src/meetnet/droogtemeetnet_script.R:98:  habmap_terr <- read_habitatmap_terr(file = "20_processed/habitatmap_terr/habitatmap_terr_inclbos.gpkg")
> ```

Thanks for this remark. Once again useful, helpful and to the point.
I added a download link.
I hope and I'm curious you'll become the same results as I did.

florisvdh committed:
Inleiding: fixed path of gw_types

florisvdh commented:
Beside a small fix in #10, I now met the following:

```
Quitting from lines 1126-1153 (droogtemeetnet_ontwerp.Rmd) 
Error in grid.newpage() : 
  could not open file 'G:/Mijn Drive/PRJ_Meetnet_Droogte/2_Uitvoering/figuren/habmap_gw_raster_overlay.png'
Calls: <Anonymous> ... print.tmap -> print_tmap -> prepare_vp -> grid.newpage
Please delete droogtemeetnet_ontwerp.Rmd after you finish debugging the error.
Execution halted
```

Will stop here for now :wink: . Some aspects of reproducibility (but not the above) may be checked on your PC as well, by cloning the repo in a second, independent folder, and trying to compile again.

w-jan committed:
Merge pull request #10 from inbo/update_fv

Inleiding: fixed path of gw_types

w-jan commented:
> Beside a small fix in #10, I now met the following:
> 
> ```
> Quitting from lines 1126-1153 (droogtemeetnet_ontwerp.Rmd) 
> Error in grid.newpage() : 
>   could not open file 'G:/Mijn Drive/PRJ_Meetnet_Droogte/2_Uitvoering/figuren/habmap_gw_raster_overlay.png'
> Calls: <Anonymous> ... print.tmap -> print_tmap -> prepare_vp -> grid.newpage
> Please delete droogtemeetnet_ontwerp.Rmd after you finish debugging the error.
> Execution halted
> ```
> 
> Will stop here for now  . Some aspects of reproducibility (but not the above) may be checked on your PC as well, by cloning the repo in a second, independent folder, and trying to compile again.

Floris,
A bit strange that you didn't encountered this so far.
It has to do with the parameter-settings. You probably choosed for params:
  refresh_data: 2
  refresh_figures: 2
With these you'll have to give up a local directory where one can save the produced data and figures :
see datapath in 00_setup (r.46) and figpath (r. 53).
if you choose refresh_data=1 and refresh_figures=1 then you don't to do this. You use the data saved in the cloud. It's also an option for a review.

w-jan committed:
small update 00_setup

w-jan committed:
Merge branch 'Versie2020_05' of https://github.com/inbo/n2khab-droogtemeetnet into Versie2020_05

florisvdh committed:
renv.lock update *

renv.lock contained packages, not used in the project and hence superfluous.
Cf. the output of renv::clean():

> renv::clean()
* No stale lockfiles were found.
* No temporary directories were found in the project library.
* No non-system packages were discovered in the system library.
The following packages are installed in the project library,
but appear to be no longer used in your project.

	car, carData, fitdistrplus, foreach, INLA, lme4, maptools,
	MatrixModels, minqa, nloptr, openxlsx, pbkrtest, quantreg,
	RcppEigen, rio, SparseM, statmod, zip

These packages will be removed.

Do you want to proceed? [y/N]: y
* Removing package(s) from project library ...
Removing package 'car' ... Done!
Removing package 'carData' ... Done!
Removing package 'fitdistrplus' ... Done!
Removing package 'foreach' ... Done!
Removing package 'INLA' ... Done!
Removing package 'lme4' ... Done!
Removing package 'maptools' ... Done!
Removing package 'MatrixModels' ... Done!
Removing package 'minqa' ... Done!
Removing package 'nloptr' ... Done!
Removing package 'openxlsx' ... Done!
Removing package 'pbkrtest' ... Done!
Removing package 'quantreg' ... Done!
Removing package 'RcppEigen' ... Done!
Removing package 'rio' ... Done!
Removing package 'SparseM' ... Done!
Removing package 'statmod' ... Done!
Removing package 'zip' ... Done!
* Done! Removed 18 packages.
* The project has been cleaned.

florisvdh committed:
00_setup: allow gargle to use cached tokens *

From https://gargle.r-lib.org/articles/non-interactive-auth.html:
"you need to successfully authenticate once in an interactive session and then, in your code, give gargle permission to use a token it finds in the cache."

florisvdh committed:
Create directory data/result/meetnet/GIS

florisvdh committed:
gitignore: exclude G: folder

florisvdh commented:
@w-jan here's the further story.

### Parameters at value 2

Regarding the error:

```
Error in grid.newpage() : 
  could not open file 'G:/Mijn Drive/PRJ_Meetnet_Droogte/2_Uitvoering/figuren/habmap_gw_raster_overlay.png'
```
... indeed created with parameters set at following values - you even made that the default for RStudio's 'build' button in commit fb32f89 so I think it should 'able to run' reproducibly (being able to regenerate data and figs is the ultimate idea after all):

- refresh_data: 2
- refresh_figures: 2

Up to that error, this is how my "`G:`" folder looks like because of the foregoing code that had run:

```bash
$ tree G\:/
G:/
└── Mijn Drive
    └── PRJ_Meetnet_Droogte
        └── 2_Uitvoering
            └── data
                └── GIS
                    └── VoorR
                        ├── grts_level0.tif
                        ├── grts_level5.tif
                        ├── grts_level8.tif
                        ├── habmap_gw_raster_overlay.gpkg
                        ├── habmap_terr_gw.gpkg
                        └── raster_meetnet_poly.gpkg

6 directories, 6 files
```
Error comes from below chunk. Probably it is because the `figpath` is never actually created by the code (while `datapath/GIS/VoorR`, is created in 00_setup), cf. above dir structure, and so the error probably comes from `tmap_save()`.

```r
if (params$refresh_figures == 2) {
  habmap_gw_raster_overlay_tm <- raster_meetnet_poly_tm + 
    tm_shape(habmap_gw_raster_overlay) + 
    tm_fill(col = "groupnr", style = "cat", palette = "BuGn", title = "Grondwatertype") + 
    tm_layout(title = "Verspreiding van de GT-groepen", asp = 0 )
  
  tmap_save(habmap_gw_raster_overlay_tm, 
            filename = file.path(figpath, "habmap_gw_raster_overlay.png"),
            dpi = 250
            )
  include_graphics(path = file.path(figpath, "habmap_gw_raster_overlay.png")) 

}
```

If you like, I'm willing to drop the G: folder and rerun with value 2 for the params after you've further tweaked the code. Just to check that it would run (correctly) in absence of the needed paths (which, concerning the _datapath_ in G:, has worked well up to the error).

I have not proceeded into solving the above, but it would be best if parametervalue of 2 gives a reproducible run.

### Parameters at value 1

Currently proceeding with parameters set at non-default value 1 as you recommend :

```r
bookdown::render_book("index.Rmd", 
                      "bookdown::html_document2", 
                      params = list(refresh_data = 1,
                                    refresh_figures = 1))
```

It errors here:

```
Quitting from lines 3174-3179 (droogtemeetnet_ontwerp.Rmd) 
Error in CPL_write_ogr(obj, dsn, layer, driver, as.character(dataset_options),  : 
  Write error.
```

I.e. it can't run:

```r
write_sf(tubes_cat1_polyg, file.path("data", "result", "meetnet", "GIS", "bijkomende_hokken.shp"))
write_sf(tubes_cat1_polyg, file.path("data", "result", "meetnet", "GIS", "bijkomende_hokken.gpkg")) 
```
Probably another problem with directory 'GIS' that should have been created first? See:

```bash
$ tree data/result/
data/result/
└── meetnet
    ├── tubes_reserve.tsv
    ├── tubes_reserve.yml
    ├── tubes_selected.tsv
    └── tubes_selected.yml

1 directory, 4 files
```

After adding commit fdf5ac9, the whole run was successful :smile_cat:  - **[here](https://github.com/inbo/n2khab-droogtemeetnet/files/5119554/droogtemeetnet_ontwerp.html.zip)** is the zipped html so that you can verify & compare it - and now we see:

```
$ tree data/result/
data/result/
└── meetnet
    ├── GIS
    │   ├── bijkomende_hokken.dbf
    │   ├── bijkomende_hokken.gpkg
    │   ├── bijkomende_hokken.prj
    │   ├── bijkomende_hokken.shp
    │   ├── bijkomende_hokken.shx
    │   ├── tubes_reserve_2020_05.dbf
    │   ├── tubes_reserve_2020_05.gpkg
    │   ├── tubes_reserve_2020_05.prj
    │   ├── tubes_reserve_2020_05.shp
    │   ├── tubes_reserve_2020_05.shx
    │   ├── tubes_selected_2020_05.dbf
    │   ├── tubes_selected_2020_05.gpkg
    │   ├── tubes_selected_2020_05.prj
    │   ├── tubes_selected_2020_05.shp
    │   └── tubes_selected_2020_05.shx
    ├── tubes_reserve.tsv
    ├── tubes_reserve.yml
    ├── tubes_selected.tsv
    └── tubes_selected.yml

2 directories, 19 files
```

Beside that, I saw a relevant warning after my first try with parameters at value 1: `In if (uitgesloten_tubes != "leeg") { ... :   the condition has length > 1 and only the first element will be used`, referring to below chunk:

```r
#peilbuizen uit de lijst verwijderen die niet (meer) mogen meegenomen worden
if (uitgesloten_tubes != "leeg"){
  tubes_hab <- tubes_hab[!(tubes_hab$loc_code %in% uitgesloten_tubes),]
}
```

Apparently `uitgesloten_tubes` is a vector with length > 1 - best investigate and solve it. It may be harmless, but it could harm. Anyway a vector with length > 1 was clearly not intended.

florisvdh committed:
Use 'cache' chunk option from parameters ('FALSE' for reproducible building)

florisvdh commented:
@w-jan regarding reproducibility, and the fact that you chose a default of value '2' for parameters `refresh_data` and `refresh_figures` (i.e. maximizing reproducibility), I have suggested (in e7e7f6a) to pass the `cache` chunk option as parameters instead of hard-code it, so that no hard-coded `cache=TRUE` appears anymore in the code. In line with your parameter settings of `refresh_data` and `refresh_figures`, I've set those new ones to `FALSE` as a default - I believe that is necessary if you want the default setting to be a reproducible build (i.e. redoing all calculations).

See https://bookdown.org/yihui/rmarkdown/parameterized-reports.html if you want to read into this.

There was one general setting in 00_setup, and in two specific R-chunks `cache=TRUE` was specifically set. As I don't know the specific reasons for the latter, I have made distinct parameters of that (so you can now build the project e.g. with a general `cache=FALSE` but with the two specific chunks still set as `TRUE`, just by passing those parameters to `bookdown::render_book()`. If you think it wasn't necessary to distinguish those specific chunk options, you can just discard those chunk settings and the corresponding parameters (see the commit to find them).

Note, it should never be needed to tinker with the code itself in order to run a build with different 'environment' settings (this may also hold for filepaths), all that should happen with parameters if others (incl. your future self) are to use it easily later on.

So, feel free to further change this to your likes.

w-jan committed:
Merge pull request #11 from inbo/updates_fv

Updates and fixes

w-jan commented:
@florisvdh 
About the run-options I dive into the code again, because frankly the logic of some choices has become unclear to me

w-jan committed:
changing the options to execute/refresh the data
refresh-options 3, 2 or 1: full versions
3: make use of the most recent files on Google Drive
2: full calculation with local files,
1: full calculation except for two time-consuming chunks: here a cached-version will be used instead
0: calcalution with use of intermediary calculated files (and the cached version of two chunks)

w-jan committed:
result of a run on 2020.09.03
no substantial differences compared with the version of 2020.05

w-jan commented:
@florisvdh 
Thanks for raising the issue of non-reproducible run-options.
The execute-options are changed. I would hope that now the result is reproducible.

florisvdh commented:
@w-jan you can quite easily test that by cloning this repo in a new location on your PC, and running that bookdown project instead - it would be a clean copy without lingering git-ignored files.

`git clone --branch Versie2020_05 https://github.com/inbo/n2khab-droogtemeetnet.git testfolder`

w-jan committed:
update after testing with cloned copies

w-jan committed:
Small update (thought this was already made)

w-jan committed:
droogtemeetnet_script.R: small update

w-jan committed:
update libraries : renv::snapshot
- update R-version -> 4.0.2
- MASS and mgcv
A bit strange: the hash of n2khab changed without changing remotesha. This library needed to be re-installed after using the new R-version.

w-jan committed:
rerun after R-version update
